packages/components/src/Button.tsx

```diff
@@ -1,27 +1,49 @@
-import type { ButtonVariants } from './styles/Button.css';
+import type { VariantProps } from 'class-variance-authority';
 import type { ForwardedRef } from 'react';
 import type { ButtonProps as AriaButtonProps } from 'react-aria-components';
 
-import { clsx } from 'clsx';
+import { cva } from 'class-variance-authority';
 import { forwardRef } from 'react';
 import { Button as AriaButton } from 'react-aria-components';
 
-import { variants } from './styles/Button.css';
+import styles from './styles/Button.module.css';
 
+const button = cva(styles.base, {
+  variants: {
+    size: {
+      small: styles.small,
+      medium: styles.medium,
+      large: styles.large,
+    },
+    variant: {
+      default: styles.default,
+      primary: styles.primary,
+      destructive: styles.destructive,
+      minimal: styles.minimal,
+      primaryFlair: styles.primaryFlair,
+      defaultFlair: styles.defaultFlair,
+      minimalFlair: styles.minimalFlair,
+    },
+  },
+  defaultVariants: {
+    size: 'medium',
+    variant: 'default',
+  },
+});
```

Comment:
[cva](https://cva.style/docs) provides us a similar variant management DX that VE recipes did